src/js/d3ViewComponent.js

```diff
@@ -0,0 +1,97 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // The D3 view component that is able to:
+    // 1. convert jQuery DOM elements to D3 elements;
+    // 2. register D3 DOM event listeners;
+    // 3. Synthesize that.options.styles and that.options.selectors to combine elements with the same key into that.classes
+
+    fluid.defaults("gpii.d3ViewComponent", {
+        gradeNames: ["fluid.viewComponent", "autoInit"],
+        members: {
+            classes: {
+                expander: {
+                    funcName: "gpii.d3ViewComponent.synthesizeClasses",
+                    args: ["{that}.options.styles", "{that}.options.selectors"]
+                }
+            }
+        },
+        invokers: {
+            jQueryToD3: {
+                funcName: "gpii.d3.jQueryToD3",
+                args: ["{arguments}.0"]
+            },
+            addD3Listeners: {
+                funcName: "gpii.d3.addD3Listeners",
+                args: ["{arguments}.0", "{arguments}.1", "{arguments}.2", "{that}"]
+            }
+        }
+    });
+
+    // Validate the given selector to ensure it is indeed of the form "period plus classname" since the element
+    // is achieved through adding a classname via the d3 "class" directivethe and the current implementation
+    // couldn't handle any other form.
+    gpii.d3ViewComponent.extractSelectorName = function (selector) {
+        if (!selector) {
+            return;
+        }
+        selector = selector.trim();
+        if (gpii.isCssClass(selector)) {
```

Comment:
`gpii.isCssClass` doesn't seem to handle the case where the selector is a set of class names . For example.

```javascript
selectors: {
    one: ".foo.bar"
    two: ".foo .baz" 
}
```

Would either of those be valid use cases? It might also help if the comment specified why it can't handle things that aren't classes.